src/commands/mail/deliverybox/create.tsx

```diff
@@ -0,0 +1,153 @@
+import { Text } from "ink";
+import { Success } from "../../../rendering/react/components/Success.js";
+import { ReactNode } from "react";
+import { ProcessRenderer } from "../../../rendering/process/process.js";
+import { FlagInput, OutputFlags } from "@oclif/core/lib/interfaces/parser.js";
+import crypto from "crypto";
+import { Value } from "../../../rendering/react/components/Value.js";
+
+type CreateResult = {
+  deliveryboxId: string;
+  generatedPassword: string | null;
+};
+
+export default class Create extends ExecRenderBaseCommand<
+  typeof Create,
+  CreateResult
+> {
+  static summary = "Create a new mail deliverybox";
+  static description = `\
+    This command can be used to create a new mail deliverybox in a project.
+  
+    When running this command with the --quiet flag, the output will contain the ID of the newly created deliverybox.
+    In addition, when run with --generated-password the output will be the ID of the newly created deliverybox, followed by a tab character and the generated password.`;
+  static flags = {
+    ...projectFlags,
+    ...processFlags,
+    description: Flags.string({
+      char: "d",
+      summary: "mail devliverybox description",
+      required: true,
+    }),
+    password: Flags.string({
+      summary: "deliverybox password",
+      description:
+        "This is the password that should be used for the deliverybox; if omitted, the command will prompt interactively for a password.\n\nCAUTION: providing this flag may log your password in your shell history!",
+    }),
+    "random-password": Flags.boolean({
+      summary: "generate a random password",
+      description:
+        "This flag will cause the command to generate a random 32-character password for the deliverybox; when running with --quiet, the devliverybox ID and the password will be printed to stdout, separated by a tab character.",
+    }),
+  };
+
+  static examples = [
+    {
+      description: "Create non-interactively with password",
+      command:
+        "$ read -s PASSWORD &&\n <%= config.bin %> <%= command.id %> --password $PASSWORD --description my personal deliverybox",
+    },
+    {
+      description: "Create non-interactively with random password",
+      command:
+        "<%= config.bin %> <%= command.id %> --random-password --description my personal deliverybox",
+    },
+  ];
+
+  protected async getPassword(
+    process: ProcessRenderer,
+  ): Promise<[string, boolean]> {
+    if (this.flags.password) {
+      return [this.flags.password, false];
+    }
+
+    if (this.flags["random-password"]) {
+      const generated = await process.runStep(
+        "generating random password",
+        async () => {
+          return crypto.randomBytes(32).toString("base64").substring(0, 32);
+        },
+      );
+
+      process.addInfo(
+        <Text>
+          generated password: <Value>{generated}</Value>
+        </Text>,
+      );
+      return [generated, true];
+    }
```

Comment:
This (together with a few other bits and pieces in this class) is probably a duplication of the `mail address create` command? Can we extract the duplicated code to a commonly-used function?